components/renku_data_services/storage/rclone_patches.py

```diff
@@ -0,0 +1,264 @@
+"""Patches to apply to phe rclone storage schema."""
+
+from copy import deepcopy
+from typing import Any
+
+from renku_data_services import errors
+
+BANNED_STORAGE = {
+    "alias",
+    "crypt",
+    "cache",
+    "chunker",
+    "combine",
+    "compress",
+    "hasher",
+    "local",
+    "memory",
+    "union",
+}
+
+OAUTH_PROVIDERS = [
+    "acd",
+    "box",
+    "drive",
+    "dropbox",
+    "gcs",
+    "gphotos",
+    "hidrive",
+    "jottacloud",
+    "mailru",
+    "onedrive",
+    "pcloud",
+    "pikpak",
+    "premiumzeme",
+    "putio",
+    "sharefile",
+    "yandex",
+    "zoho",
+]
+
+
+def find_storage(spec: list[dict[str, Any]], prefix: str) -> dict[str, Any]:
+    """Find and return the storage schema from the spec.
+
+    This returns the original entry for in-place modification.
+    """
+    storage = next((s for s in spec if s["Prefix"] == prefix), None)
+    if not storage:
+        raise errors.ValidationError(message=f"'{prefix}' storage not found in schema.")
+    return storage
+
+
+def patch_schema_remove_unsafe(spec: list[dict[str, Any]]) -> list[dict[str, Any]]:
```

Comment:
question: I am being really nitpicky here. But should we just not return anything from any of these functions? Because they all modify the spec in place as far as I can tell. So there is really no need to return anything. We can also keep copying and then return stuff OR we dont return anything and modify in place OR we at least mention that we modify in place in the doc strings. But I personally think that just not returning anything makes it really clear what is going on.